server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -525,6 +528,7 @@ protected Node(
                     b.bind(ScriptService.class).toInstance(scriptModule.getScriptService());
                     b.bind(AnalysisRegistry.class).toInstance(analysisModule.getAnalysisRegistry());
                     b.bind(IngestService.class).toInstance(ingestService);
+                    b.bind(ConsistentSecureSettingsValidatorService.class).toInstance(consistentSecureSettingsValidatorService);
```

Comment:
why do need to bind this?

Reply:
I don't have a definite idea about how users of the consistent-secure-setting feature would pertain an instance of this. Could it be a field in `ClusterService`? That was my original thinking... that does not sound like an improvement over this. Now that the approach has not been vetoed I will think more about exposing it to clients. Any suggestions are welcome.

Reply:
I wonder if rather than a field in the cluster service if it could be a method? That way a client can just ask if setting X is consistent and not need to worry about how that is handled internally?

We want to remove Guice and everything that gets bound is another hurdle we have to deal with.

Reply:
> I wonder if rather than a field in the cluster service if it could be a method?

Absolutely, the interface should support a `Setting` object, I got that. The method though, needs access to the `ConsistentSecureSettingsValidatorService` (be it a private field).
One obstacle to this , is that `ConsistentSecureSettingsValidatorService` needs a `ClusterService` field, and , with this proposition, the `ClusterService` needs a `ConsistentSecureSettingsValidatorService` field...